src/custom/components/Tokens/BalanceCell.tsx

```diff
@@ -0,0 +1,25 @@
+import { Token, CurrencyAmount } from '@uniswap/sdk-core'
+import { useActiveWeb3React } from 'hooks/web3'
+import { BalanceValue } from './styled'
+import { formatSmart } from 'utils/format'
+import Loader from 'components/Loader'
+
+type BalanceCellProps = {
+  balance: CurrencyAmount<Token> | undefined
+}
+
+export default function BalanceCell({ balance }: BalanceCellProps) {
+  const { account } = useActiveWeb3React()
+  const hasBalance = balance?.greaterThan(0)
+  const formattedBalance = formatSmart(balance) || 0
+
+  if (!balance) {
+    return account ? <Loader /> : <BalanceValue hasBalance={false}>0</BalanceValue>
+  }
+
+  return (
+    <BalanceValue title={balance?.toExact()} hasBalance={!!hasBalance}>
```

Comment:
Nitpick: at this point we know the balance is defined

```suggestion
    <BalanceValue title={balance.toExact()} hasBalance={!!hasBalance}>
```

Also for the `title` text [there's `formatMax`](https://github.com/cowprotocol/cowswap/blob/9ee39d414b760e67f36335c5ac46c517e1106d92/src/custom/utils/format.ts#L164=)